src/main/java/uk/gov/hmcts/reform/orgrolemapping/data/AccessTypesRepository.java

```diff
@@ -0,0 +1,17 @@
+package uk.gov.hmcts.reform.orgrolemapping.data;
+
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface AccessTypesRepository extends CrudRepository<AccessTypes, Long> {
+
+    //retrieves the AccessTypes record by version.
+    AccessTypes findByVersion(Long version);
+
+    // Retrieves the first AccessTypes record.
+    // 1. It first sorts the AccessTypes records in descending order by version.
+    // 2.Hence, we get the top version among all the AccessTypes
+    AccessTypes findFirstByOrderByVersionDesc();
```

Comment:
`findFirstByOrderByVersionDesc()` is sorting required, if there will only be 1 row? see: https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1723368280#PRMMVP-3.7.5.2CaseDefinitionAccessTypes `findFirst()` should be fine?

Reply:
There are lots of rows in the table versioned. So to return only the last version, yes it is necessary to sort so the latest is the first row and that is what gets returned. See the ticket as I have added a manual test of this.